packages/js/settings-editor/src/legacy/form.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { createElement, useRef } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { DataForm } from '@wordpress/dataviews';
+import { getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { useSettingsForm } from '../hooks/use-settings-form';
+import { CustomView } from '../components/custom-view';
+
+export const Form = ( {
+	settings,
+	settingsData,
+	settingsPage,
+	activeSection,
+}: {
+	settings: SettingsField[];
+	settingsData: SettingsData;
+	settingsPage: SettingsPage;
+	activeSection: string;
+} ) => {
+	const { data, fields, form, updateField } = useSettingsForm( settings );
+	const formRef = useRef< HTMLFormElement >( null );
+
+	const getFormData = () => {
+		if ( ! formRef.current ) {
+			return {};
+		}
+		const formElements = formRef.current.querySelectorAll<
+			HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement
+		>( 'input, select, textarea' );
+
+		const formData: Record< string, string > = {};
+
+		formElements.forEach( ( input ) => {
+			// Avoid generic Gutenberg input ids. This will require upstream fixes.
+			if ( input.id.startsWith( 'inspector-' ) ) {
+				return;
+			}
+
+			formData[ input.name || input.id ] = input.value;
+		} );
+
+		return formData;
+	};
+
+	const handleSubmit = ( event: React.FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		const query: Record< string, string > = {
+			page: 'wc-settings',
+			tab: settingsPage.slug,
+		};
+		if ( activeSection !== 'default' ) {
+			query.section = activeSection;
+		}
+
+		const formData = getFormData();
+		formData.save = 'Save changes';
+		formData._wpnonce = settingsData._wpnonce;
+		formData._w_http_referer = '/wp-admin/' + getNewPath( query );
+
+		// eslint-disable-next-line no-console
+		console.log(
```

Comment:
Do we want a console.log here when its merged?

Reply:
Yes, I think so. The follow up PRs depend on it https://github.com/woocommerce/woocommerce/pull/55466

But its only temporary, I'll remove it when creating the API https://github.com/woocommerce/woocommerce/pull/55467